### PR TITLE
Adds Meson Sunglasses

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do
Adds Meson Sunglasses to the game, with relevant crafting recipes and adds them to the loadout of the Quartermaster and Garment Bag of the Chief Engineer.

## Why It's Good For The Game
Consistency with other types of HUDs. Gives the QM/CE more consistency with other Heads of Staff roles who have sunglasses variants of their department HUDs. Makes engineers who get sunglasses even drippier.

## Images of changes
image
image
image

## Testing
Logged in as QM, spawned a CE garment bag, checked its contents. Took my glasses off, turned them back into sunglasses and mesons. Got some cable and turned them back into mesonHUD sunglasses.

## Changelog
:cl:
add: Added meson sunglasses to the game.
tweak: The Quartermaster now spawns with a pair of meson sunglasses on.
tweak: The Chief Engineer's garment bag now contains a pair of meson sunglasses.
/:cl:

### CODE REVIEW

**Comment by Danchi299** [comment]
the only missing combinations now should be Janitorial Goggles and Skills Implant

File: code/modules/crafting/tailoring.dm [hunk 1]
```diff
@@ -212,12 +212,29 @@
...
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	reqs = list(/obj/item/clothing/glasses/meson/sunglasses = 1)
+	category = CAT_CLOTHING
 /datum/crafting_recipe/beergoggles
 	name = "Sunscanners"
 	result = list(/obj/item/clothing/glasses/sunglasses/reagent)
 	time = 20
 	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
-	reqs = list(/obj/item/clothing/glasses/science = 1,
+	reqs = list(/obj/item/clothing/glasses/meson = 1,
```
Comment (1080pCat) [inline code/modules/crafting/tailoring.dm:237]: 🤨

Comment (ItsMarmite) [reply code/modules/crafting/tailoring.dm:237]: Why do you want me to make the bartender's sunscanners craftable with mesons?

Comment (ItsMarmite) [reply code/modules/crafting/tailoring.dm:237]: I'm an idiot hold on.

File: code/modules/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -119,6 +119,14 @@
 	see_in_dark = 8
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 
+/obj/item/clothing/glasses/meson/sunglasses
+	name = "meson HUD sunglasses"
+	desc = "Sunglasses with an inbuilt scanner that can see through both walls and flooring.."
```
Comment (1080pCat) [inline code/modules/clothing/glasses/glasses.dm:124]:
```suggestion
	desc = "Sunglasses with an inbuilt scanner that can see through both walls and flooring."
```

File: code/modules/clothing/glasses/glasses.dm [hunk 3]
```diff
@@ -119,6 +119,14 @@
...
+	icon_state = "sunhudmeson"
+	see_in_dark = 1
```
Comment (1080pCat) [inline code/modules/clothing/glasses/glasses.dm:126]: sunglasses giving a low light buff is a little odd tbh.
unless im misinterpreting this

**Review by 1080pCat — COMMENTED** [review]
couple more things i noticed on second pass

File: code/modules/clothing/glasses/glasses.dm (hunk 3 above)
Comment (ItsMarmite) [reply code/modules/clothing/glasses/glasses.dm:126]: Just a leftover from copypasting janihud sunglasses over for the format, I assume that was on it to made the janihud itself look better. I'll toss it.

**Review by 1080pCat — APPROVED** [review]

File: code/game/objects/items/weapons/storage/garment.dm [hunk 4]
```diff
@@ -138,6 +138,7 @@
 	new /obj/item/clothing/shoes/brown(src)
 	new /obj/item/clothing/glasses/welding/superior(src)
 	new /obj/item/clothing/gloves/color/yellow(src)
+	new /obj/item/clothing/glasses/meson/sunglasses(src)
```
Comment (Henri215) [inline code/game/objects/items/weapons/storage/garment.dm:141]:
```suggestion
	new /obj/item/clothing/glasses/welding/superior(src)
	new /obj/item/clothing/glasses/meson/sunglasses(src)
	new /obj/item/clothing/gloves/color/yellow(src)
```
glasses next to glasses

File: code/modules/crafting/tailoring.dm [hunk 5]
```diff
@@ -212,6 +212,23 @@
...
+	category = CAT_CLOTHING
+
+/datum/crafting_recipe/hudsunmesonremoval
+	name = "Meson HUD sunglasses removal"
+	result = list(/obj/item/clothing/glasses/sunglasses, /obj/item/clothing/glasses/meson)
+	time = 20
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	reqs = list(/obj/item/clothing/glasses/meson/sunglasses = 1)
+	category = CAT_CLOTHING
 /datum/crafting_recipe/beergoggles
```
Comment (Henri215) [inline code/modules/crafting/tailoring.dm:232]:
```suggestion
	category = CAT_CLOTHING

/datum/crafting_recipe/beergoggles
```

File: code/modules/crafting/tailoring.dm [hunk 6]
```diff
@@ -212,6 +212,24 @@
 	reqs = list(/obj/item/clothing/glasses/hud/janitor/sunglasses = 1)
 	category = CAT_CLOTHING
 
+/datum/crafting_recipe/hudsunmeson
+	name = "Meson HUD sunglasses"
+	result = list(/obj/item/clothing/glasses/meson/sunglasses)
+	time = 20
```
Comment (DGamerL) [inline code/modules/crafting/tailoring.dm:218]:
```suggestion
	time = 2 SECONDS
```

File: code/modules/crafting/tailoring.dm [hunk 7]
```diff
@@ -212,6 +212,24 @@
...
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	reqs = list(/obj/item/clothing/glasses/meson = 1,
+				/obj/item/clothing/glasses/sunglasses = 1,
+				/obj/item/stack/cable_coil = 5)
+	category = CAT_CLOTHING
+
+/datum/crafting_recipe/hudsunmesonremoval
+	name = "Meson HUD sunglasses removal"
+	result = list(/obj/item/clothing/glasses/sunglasses, /obj/item/clothing/glasses/meson)
+	time = 20
```
Comment (DGamerL) [inline code/modules/crafting/tailoring.dm:228]:
```suggestion
	time = 2 SECONDS
```

**Review by Henri215 — APPROVED** [review]

**Comment by Danchi299** [comment]
I just realized that this + experimental welder is engineer's best friend, totally adding this to CE locker instead of superior welding goggles

**Review by S34NW — APPROVED** [review]

**Pull request merged**